util/topology/cluster.go

```diff
@@ -76,6 +77,18 @@ func GetMDVariable(cluster *clusterv1.Cluster, mdIndex int, name string, value i
 	return errors.Wrap(err, "unmarshalling from JSON into value")
 }
 
+// IsSingleNodeCluster checks if the cluster topology is single node cluster(with CP count as 1 and worker count as 0).
+// Pre-reqs: cluster.Spec.Topology != nil
+func IsSingleNodeCluster(cluster *clusterv1.Cluster) bool {
+	return *cluster.Spec.Topology.ControlPlane.Replicas == *pointer.Int32(1) && HasWorkerNodes(cluster)
+}
+
+// HasWorkerNodes checks if the cluster topology has worker nodes.
+// Pre-reqs: cluster.Spec.Topology != nil
+func HasWorkerNodes(cluster *clusterv1.Cluster) bool {
```

Comment:
HasWorkerNodes->NoWorkerNodes

Reply:
since its a bool returning func, doesn't HasWorkerNodes makes more sense?

Reply:
Then it is better to use !HasWorkerNodes in the references.